test/dlc_tests/conftest.py

```diff
@@ -363,6 +369,7 @@ def efa_ec2_instances(
     region,
     availability_zone_options,
 ):
+    _restrict_c4_usage(ec2_instance_type)
```

Comment:
nit: why not just call this from within the ec2_instance_type fixture itself, rather than first line of function calls?